querytest/compare.go

```diff
@@ -0,0 +1,222 @@
+package querytest
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/values"
+)
+
+// FloatTolerance represents a sufficient distance that all
+// float values must be within in order to be considered equal.
+const FloatTolerance float64 = 1e-50
+
+// FloatULP is the maximum number of floats that are allowed to lie
+// between two floating point values and still be considered equal.
+const FloatULP uint = 2
+
+type tables struct {
+	name string
+	data flux.ResultIterator
+}
+
+func CreateTables(name string, data flux.ResultIterator) *tables {
+	return &tables{
+		name: name,
+		data: data,
+	}
+}
+
+type results struct {
+	equal bool
+	err   error
+	fmt   map[string]*execute.Formatter
+}
+
+func (r *results) Equal() bool {
+	return r.equal
+}
+func (r *results) Err() error {
+	return r.err
+}
+func (r *results) Fmt() map[string]*execute.Formatter {
+	return r.fmt
+}
+
+func (t *tables) EqualWithTolerance(s *tables, tol float64, ulp uint) *results {
+	defer t.data.Release()
+	defer s.data.Release()
+	for t.data.More() && s.data.More() {
+		l := t.data.Next()
+		r := s.data.Next()
+		if l.Name() != r.Name() {
+			return &results{err: fmt.Errorf(
+				"result names not equal - %s: %s != %s: %s",
+				t.name, l.Name(), s.name, r.Name()),
+			}
+		}
+		var lt, rt []table
```

Comment:
Yeah you're right I need better names. I just couldn't think of good descriptions. The reason tables have a Name param is so that the caller can label which result/table corresponds to "want" and which one corresponds to "got".